### PR TITLE
Revert async nameresolution on Unix

### PR DESCRIPTION
**Pull request by gfoidl**
As per discussion in https://github.com/dotnet/runtime/issues/48566#issuecomment-784484616

In short: `getaddrinfo_a` is a sub-optimal api, that doesn't perform real async name resolution. Other ways will be explored.

This PR undos https://github.com/dotnet/runtime/pull/34633 and https://github.com/dotnet/runtime/pull/47171

### CODE REVIEW

File: src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs [hunk 1]
```diff
@@ -547,6 +547,8 @@ private static Task GetHostEntryOrAddressesCoreAsync(string hostName, bool justR
             {
                 if (NameResolutionPal.SupportsGetAddrInfoAsync)
                 {
+#pragma warning disable CS0162 // Unreachable code detected -- SupportsGetAddrInfoAsync is a constant on *nix.
```
Comment (stephentoub) [inline src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs:550]: Where is the warning restored?

Comment (gfoidl) [reply src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs:550]: The async tests didn't exist before #34633, but I'd like to keep them as they test the Windows-PAL / everything where `NameResolutionPal.SupportsGetAddrInfoAsync = true`.

Comment (stephentoub) [reply src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs:550]: What I meant was, where is the corresponding `#pragma warning restore CS0162`?  I don't see it, which would mean this warning is being disabled accidentally for the rest of the file.

Comment (gfoidl) [reply src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs:550]: I wondered this myself, as it was in the `Dns`-class without a #pragma restore, so just "copyed" this way over.
Happy that you confirmed that a restore is necessary.

Done with 8de5e3245a4a0fc02dd52b36615ac89802e002c5

**Review by jkotas — APPROVED** [review]

**Comment by jkotas** [comment]
@dotnet/ncl Good to merge?

**Review by stephentoub — APPROVED** [review]
Thanks, @gfoidl.

**Pull request merged**